client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -138,9 +138,20 @@ const Header: FunctionComponent = () => {
 		</span>
 	);
 
-	const currentDomain = domain ?? freeDomainSuggestion;
+	const [ shownDomain, setShownDomain ] = React.useState<
+		import('@automattic/data-stores').DomainSuggestions.DomainSuggestion
+	>();
+	React.useEffect( () => {
+		if ( domain && shownDomain !== domain ) {
+			setShownDomain( shownDomain );
+			return;
+		}
+		if ( ! shownDomain && freeDomainSuggestion ) {
+			setShownDomain( freeDomainSuggestion );
+		}
+	}, [ domain, shownDomain, freeDomainSuggestion ] );
```

Comment:
Had to add this to prevent domain modal from disappearing when we change the search from within the modal…

It's not ideal to track all of the domain search in the top-level onboard store. Search via vertical or site title is one thing, but text entered into the search bar or categories feels very much like component scope state.

Reply:
> to prevent domain modal from disappearing when we change the search from within the modal

Expanding: The search would change, the selector would trigger the resolver and return no results, the domain picker is unmounted until resolver resolves and the selector has results, then the domain picker button is remounted but it's internal state showing the domain picker modal has been lost.